tests/test_job.py

```diff
@@ -1476,3 +1476,22 @@ def check_child_process_exited(hq_env: HqEnv, stop_fn: Callable[[subprocess.Pope
     parent, child = pids
     wait_for_pid_exit(parent)
     wait_for_pid_exit(child)
+
+def test_job_task_ids(hq_env: HqEnv):
+    hq_env.start_server()
+    hq_env.command(["submit", "--array=2,7,9,20-30", "--", "python", "-c", "import os; assert os.environ['HQ_TASK_ID'] not in ['25', '26', '27', '28']"])
+    hq_env.start_workers(1, cpus=1)
+    wait_for_job_state(hq_env, 1, "FAILED")
```

Comment:
We should wait until all tasks have finished :thinking: Otherwise there could be a race condition in the test.

Reply:
It is not a race. FAILED state (of a job) is when there are no running & waiting tasks.

Reply:
Oh, you're right, sorry. Nevermind :)